### DISCUSSION TITLE
Is it possible to create partial matching routes? e.g. example.com/@username

### DISCUSSION THREAD

**Discussion by slowfound** [vercel/next.js#28743] [Help]
I've recently converted my CRA app to next.js. I was easily handling this route in this manner:

`<Route path="/@:username" component={ProfileScreen} />`

I want to be able to pass the `username` variable to the link component, maybe like this:

```<Link href={`/@${username}`} >
               ${username}
    </Link>```
             
 I tried creating my pages folder like this:
 
 ``` pages/@[username].js```
 
 But it didn't work. Is this possible?

**Comment by awareness481** [comment]
I would suggest adding the '@' in your getStaticPaths.

```jsx
export async function getStaticPaths() {
  const usernames = // fetch usernames etc
  
  const paths = usernames.map((username) => {
    return {
      params: '@' + username
    };
  }
  
  return {
    paths
  }
}
```

Something like this should work.
